### ISSUE TITLE
Difference between edge_index and edge_label_index

### ISSUE THREAD

**Issue by hojun0705** [snap-stanford/deepsnap#39]
Hello. I know what edge_index is but I am still struggling with edge_label_index.

Could you please make it clear with some examples and detailed explanation?

Thank you in advance.

**Comment by zechengz** [comment]
Hi,

In general, the `edge_label_index ` specifies which edges have the label. For example, if we have `edge_label_index` the `[[0, 1, 2], [1, 2, 0]]` This means that these three edges `(0, 1), (1, 2), (2, 0)` having the label. The reason why we create the `edge_label_index` is that in the task such as the link prediction, we need to know which edges should be fed to the loss function and do the backpropagation. In the example above, we can easily index the source node embeddings ( node `0, 1, 2`) and destination node embeddings (node `1, 2, 0`) by using the `edge_label_index`.

**Comment by hojun0705** [comment]
Thank you very much for the clarification. Much appreciated.
